### DISCUSSION TITLE
Entity couldn't be removed after removing the component

### DISCUSSION THREAD

**Discussion by chengenzhao** [AlmasB/FXGL#1097] [Q&A]
Hi:

I just have a problem of removing the entity, following code doesn't work properly
<img width="761" alt="Screen Shot 2021-08-26 at 11 17 49 AM" src="https://user-images.githubusercontent.com/5525436/130894988-6b23f1e5-d3ad-4d88-8582-9fab01a5eb33.png">
and if we delete the entity.removeComponent part(top 3 lines)
then entity could be removed from the world
but if we remove the component first
then the removedFromWorld method doesn't work

**Comment by AlmasB** [comment]
Hi,

I can't see anything wrong with the code above. Please can you provide a minimal example that reproduces the issue.

**Reply by chengenzhao** [reply]
I just zip the project and upload to zipshare
would you mind download the project from this url?
https://www.zipshare.com/download/eyJhcmNoaXZlSWQiOiIxM2IxN2ZkYS0zN2ZjLTQ4YjItYWZmNi02NzJlNzgzYTA0MjciLCJlbWFpbCI6ImNoZW5nZW4uemhhb0B3aGl0ZXdvb2RjaXR5LmNvbSJ9

**Reply by chengenzhao** [reply]
The problem code is in
com.whitewoodcity.battlefxgl.entity.component.components.DestroyableComponent
comment out stopEntity method
and uncomment entity.removeComponent
then start the game 
you may see the error
The Tank, Launcher could have such problem
if the player destroy the Tank, it wouldn't be remove from the world 
The player could jump and stand on the invisible tank or launcher

**Reply by chengenzhao** [reply]
There is a tank will spawned after 10 seconds of the game
so dont rush into the right 
just be patient, the Tank will rush into the left where the player stands
and you may change the spawn seconds in the class file:
com.whitewoodcity.battlefxgl.Game
onUpdate method
```java
int delay = 600;//10 seconds

  @Override
  protected void onUpdate(double tpf) {

    delay--;
    if(delay==0){
      spawn("TankM7", gameWidth-300, 400);
    }

```

**Reply by chengenzhao** [reply]
will have error like:
<img width="768" alt="F3E540C7-AB0A-4315-95BC-78491D75A69C" src="https://user-images.githubusercontent.com/5525436/130944732-7308a078-3923-401d-82ab-66186293f9dc.png">

**Reply by chengenzhao** [reply]
If there are too many enemies rush out then just comment out following codes in onUpdate method of Game.java
```java
if ((int) (Math.random() * 600) == 0) {
      switch (FXGLMath.random(0,10)) {
        case 0 -> spawn("TankM7", gameWidth-300, 100);
        case 1 -> spawn("Apache", gameWidth-300, 100);
        case 2 -> spawn("Launcher", gameWidth-300, 100);
        case 3 -> spawn("Yamato", gameWidth-300, 400);
        default -> spawn("F3",
          new SpawnData(gameWidth - 200, player.getY() - 400)
            .put("targetX", (int)player.getX())
            .put("targetY", (int)player.getY()));
      }
    }
```
